### PR TITLE
std.experimental.logger: structs and classes with @system toString methodes did not

### PR DESCRIPTION
**Pull request by burner**
used to work. This PR fixes that

### CODE REVIEW

File: std/experimental/logger/core.d [hunk 1]
```diff
@@ -3052,3 +3045,24 @@ unittest
     l.infof("log: %s", a);
     assert(l.msg == "log: 1");
 }
+
+/* structs and classes with @system or @trusted toString methodes did not
+used to work. This test checks that it does work now
+*/
```
Comment (schveiguy) [inline std/experimental/logger/core.d]:
```D
// Ensure system toString methods work
```

**Comment by burner** [comment]
@klickverbot the log functions call formattedWrite which calls toString of user defined types. that toString might not be safe or trusted

**Comment by burner** [comment]
@schveiguy @klickverbot updated

**Comment by schveiguy** [comment]
I think @klickverbot's point was that a `@trusted toString` can be called just fine from `@safe` code.

But the current commit looks fine, and doesn't really worry about that. LGTM.

**Comment by burner** [comment]
yes it makes system and trusted work.

**Comment by burner** [comment]
I see your point

**Comment by burner** [comment]
thanks

**Pull request merged**